uk.co.compucorp.civicrm.tasksassignments/tests/phpunit/CRM/Tasksassignments/BAO/TaskTest.php

```diff
@@ -1,31 +1,66 @@
 <?php
 
 use CRM_Tasksassignments_Test_BaseHeadlessTest as BaseHeadlessTest;
+use CRM_Tasksassignments_Test_Fabricator_Case as CaseFabricator;
+use CRM_Tasksassignments_Test_Fabricator_CaseType as CaseTypeFabricator;
+use CRM_Tasksassignments_Test_Fabricator_Contact as ContactFabricator;
+use CRM_Tasksassignments_Test_Fabricator_Task as TaskFabricator;
+use CRM_Tasksassignments_Test_Fabricator_OptionValue as OptionValueFabricator;
 
 /**
  * @group headless
  */
 class CRM_Tasksassignments_BAO_TaskTest extends BaseHeadlessTest {
-  private $_taskTypeId = null;
+  const SAMPLE_TASKS_COUNT = 3;
+
+  private $_case = null;
+  private $_taskType = null;
+  private $_tasks = [];
 
   function setUp() {
     parent::setUp();
     $upgrader = CRM_Tasksassignments_Upgrader::instance();
     $upgrader->install();
 
-    // Pick one of Task types.
-    $taskTypes = civicrm_api3('Task', 'getoptions', array(
-      'field' => 'activity_type_id',
-    ));
-    $this->_taskTypeId = array_shift($taskTypes['values']);
+    $this->_taskType = OptionValueFabricator::fabricateTaskType();
+
+    civicrm_api3('OptionGroup', 'create', ['name' => 'case_type_category']);
+    OptionValueFabricator::fabricate([ 'name' => 'WORKFLOW', 'option_group_id' => 'case_type_category' ]);
+    OptionValueFabricator::fabricate([ 'name' => 'VACANCY', 'option_group_id' => 'case_type_category'  ]);
```

Comment:
These option values are supposed to be generated by civicrm-core, not sure why they are not available on the test database.